virttest/qemu_vm.py

```diff
@@ -3118,8 +3119,16 @@ def __iothread_conflict_check(params):
             devices.insert(StrDev("noshutdown", cmdline="-no-shutdown"))
 
         user_runas = params.get("user_runas")
-        if devices.has_option("runas") and user_runas:
-            devices.insert(StrDev("runas", cmdline="-runas %s" % user_runas))
+        if devices.has_option("run-with") and user_runas:
+            devices.insert(
+                StrDev(
+                    "user_runas", cmdline="-run-with user=%s" % shlex.quote(user_runas)
+                )
+            )
+        elif devices.has_option("runas") and user_runas:
+            devices.insert(
+                StrDev("user_runas", cmdline="-runas %s" % shlex.quote(user_runas))
```

Comment:
Hi @maxujun 
Can you give more about the intention of the `shlex.quote(user_runas)`?
Thanks.